e2e/framework/applications.go

```diff
@@ -23,57 +23,59 @@ import (
+	nodes := Cluster.Provisioner().NodePool().AllocatedNodes()
+	if len(nodes) == 0 {
+		Failf("expected active nodes in cluster, got none")
+	}
 
-		nodes := Cluster.Provisioner().NodePool().AllocatedNodes()
-		if len(nodes) == 0 {
-			Failf("expected active nodes in cluster, got none")
-		}
+	stateDir := fmt.Sprintf("--state-dir=%v", TestContext.StateDir)
+	opsURL := fmt.Sprintf("--ops-url=%v", TestContext.OpsCenterURL)
+	outputPath := filepath.Join(TestContext.StateDir, "app.tar.gz")
+	cmd := exec.Command("gravity", "--insecure", stateDir, "package", "export",
+		opsURL, TestContext.Application.String(), outputPath)
+	Expect(system.Exec(cmd, os.Stderr)).To(Succeed())
+
+	versionS := TestContext.Application.Version
+	if versionS == latestMetaversion {
+		var err error
+		versionS, err = getResourceVersion(outputPath)
+		Expect(err).NotTo(HaveOccurred(), "expected to query application package version from tarball")
+	}
 
-		stateDir := fmt.Sprintf("--state-dir=%v", TestContext.StateDir)
-		opsURL := fmt.Sprintf("--ops-url=%v", TestContext.OpsCenterURL)
-		outputPath := filepath.Join(TestContext.StateDir, "app.tar.gz")
-		cmd := exec.Command("gravity", "--insecure", stateDir, "package", "export",
-			opsURL, TestContext.Application.String(), outputPath)
-		Expect(system.Exec(cmd, os.Stderr)).To(Succeed())
-
-		versionS := TestContext.Application.Version
-		if versionS == latestMetaversion {
-			var err error
-			versionS, err = getResourceVersion(outputPath)
-			Expect(err).NotTo(HaveOccurred(), "expected to query application package version from tarball")
-		}
+	version, err := semver.NewVersion(versionS)
+	Expect(err).NotTo(HaveOccurred(),
+		fmt.Sprintf("expected a version in semver format, got %q", TestContext.Application.Version))
+
+	bumpedVersion := fmt.Sprintf("--version=%v", bump(*version))
+	// Import the same package with a new version to emulate update
+	cmd = exec.Command("gravity", "--insecure", stateDir, "app", "import", opsURL, bumpedVersion, outputPath)
+	Expect(system.Exec(cmd, os.Stderr)).To(Succeed())
 
-		version, err := semver.NewVersion(versionS)
-		Expect(err).NotTo(HaveOccurred(),
-			fmt.Sprintf("expected a version in semver format, got %q", TestContext.Application.Version))
+	Distribute("gravity update", nodes[0])
+}
 
-		bumpedVersion := fmt.Sprintf("--version=%v", bump(*version))
-		// Import the same package with a new version to emulate update
-		cmd = exec.Command("gravity", "--insecure", stateDir, "app", "import", opsURL, bumpedVersion, outputPath)
-		Expect(system.Exec(cmd, os.Stderr)).To(Succeed())
+// UpdateApplicationWithInstaller impements site update via installer tarball
+func UpdateApplicationWithInstaller() {
+	Expect(ConnectToOpsCenter(TestContext.OpsCenterURL, TestContext.ServiceLogin)).To(Succeed())
+	config := infra.Config{ClusterName: TestContext.ClusterName}
+	provisioner, err := provisionerFromState(config, *testState)
```

Comment:
You don't really need to initialize the provisioner again - it is done once on each start: see https://github.com/gravitational/robotest/blob/master/e2e/framework/framework.go#L110 and  https://github.com/gravitational/robotest/blob/master/e2e/framework/framework.go#L159,L162